src/test/java/nextstep/subway/favorite/application/FavoriteServiceTest.java

```diff
@@ -0,0 +1,107 @@
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.when;
+
+@DisplayName("즐겨찾기 서비스 단위테스트")
+@ExtendWith(MockitoExtension.class)
+class FavoriteServiceTest {
+    public final static String EMAIL = "yong2ss.com";
+    public final static String PASSWORD = "12341234";
+    public final static Integer AGE = 20;
+
+    private Station 청담역;
+    private Station 뚝섬유원지역;
+    private Member 회원;
+    private Favorite 즐겨찾기;
+
+    private FavoriteService favoriteService;
+
+    @Mock
+    private FavoriteRepository favoriteRepository;
+
+    @Mock
+    private MemberRepository memberRepository;
+
+    @Mock
+    private StationRepository stationRepository;
+
+    @BeforeEach
+    void init() {
+        favoriteService = new FavoriteService(favoriteRepository, new StationService(stationRepository), memberRepository);
+    }
+
+    @Test
+    @DisplayName("회원이_출발지와_도착지로_즐겨찾기를_저장하면_정상적으로_등록된다")
+    void saveFavorite() {
+        //given
+        청담역 = new Station(1L, "청담역");
+        뚝섬유원지역 = new Station(2L, "뚝섬유원지역");
+        회원 = new Member(1L, EMAIL, PASSWORD, AGE);
+
+        //when
+        when(stationRepository.findById(1L)).thenReturn(Optional.of(청담역));
+        when(stationRepository.findById(2L)).thenReturn(Optional.of(뚝섬유원지역));
+        when(memberRepository.findById(anyLong())).thenReturn(Optional.of(회원));
+        when(favoriteRepository.save(any())).thenReturn(new Favorite(1L, 회원, 청담역, 뚝섬유원지역));
+
+        FavoriteResponse response = favoriteService.saveFavorite(new LoginMember(1L, EMAIL, AGE),
+                                                                    new FavoriteRequest(1L, 2L));
+
+        //then
+        assertAll(() -> {
+            assertThat(response.getId()).isEqualTo(1L);
+            assertThat(response.getSource().getId()).isEqualTo(청담역.getId());
+            assertThat(response.getTarget().getId()).isEqualTo(뚝섬유원지역.getId());
+        });
+    }
+
+    @Test
+    @DisplayName("회원 아이디로 저장된 즐겨찾기를 조회한다.")
+    void findByMemberId() {
+        //given
+        청담역 = new Station(1L, "청담역");
+        뚝섬유원지역 = new Station(2L, "뚝섬유원지역");
+        회원 = new Member(1L, EMAIL, PASSWORD, AGE);
+        즐겨찾기 = new Favorite(1L, 회원, 청담역, 뚝섬유원지역);
+
+        //when
+        when(memberRepository.findById(anyLong())).thenReturn(Optional.of(회원));
+        when(favoriteRepository.findByMemberId(anyLong())).thenReturn(Arrays.asList(즐겨찾기));
+
+        List<FavoriteResponse> favoriteResponses = favoriteService.findByMemberId(new LoginMember(1L, EMAIL, AGE));
+
+        //then
+        assertThat(favoriteResponses).hasSize(1);
+        assertThat(favoriteResponses.stream()
+                                    .filter(favoriteResponse -> favoriteResponse.getId() == 1L)
+                                    .findFirst()
+                                    .get().getSource()
+                                    .getId()).isEqualTo(청담역.getId());
```

Comment:
여기서 필터를 사용하는건 조금 과하지 않을까요?
결과를 미리 정하고 인과관계를 만드는 느낌이 들어서요.
정상적으로 로직을 수행했다면 필요없는 부분이기도 하고, 로직을 어떻게 풀어나가느냐에 따라 불필요할 수 있다고 생각해요.

그리고 assertThat()을 사용하면 스트림처럼 체이닝 기반의 유용한 메소드를 많이 사용할 수 있어요.
아래 코드도 같은 결과를 검증하고 있으니 고려해보시기 바랍니다.

```suggestion
        assertThat(favoriteResponses).extracting("id")
                                     .containsExactly(청담역.getId());
```